packages/styled/types/base.d.ts

```diff
@@ -63,10 +63,18 @@ export interface CreateStyledComponent<
   SpecificComponentProps extends {} = {},
   JSXProps extends {} = {}
 > {
+  (
+    template: TemplateStringsArray,
```

Comment:
I see where you are going with this reordering and I think it makes sense. However, I'm really anxious about reordering those overloads. This change needs some extra coverage in our type tests.

Reply:
Oh, I misunderstood this in part - this is actually a fix for existing tests (that became import after the change in `@emotion/serialize`)